### PR TITLE
remove unverified certs once an email is verified

### LINKED ISSUE #2924: verifying an unverified-email should remove the associated cert from localStorage

### PR DESCRIPTION
**Pull request by seanmonstar**
fixes #2924

@ozten can you make sure what I'm doing doesn't seem absoutely batty?

### CODE REVIEW

File: resources/static/common/js/user.js [hunk 1]
```diff
@@ -260,20 +260,28 @@ BrowserID.User = (function() {
...
-    _.extend(email_obj, {
-      updated: now,
-      pub: keypair.publicKey.toSimpleObject(),
-      priv: keypair.secretKey.toSimpleObject(),
-      cert: cert
-    });
+      if (info.state === "unverified") {
+        email_obj.unverified = true;
+      } else if (email_obj.unverified) {
+        delete email_obj.unverified;
```
Comment (ozten) [inline resources/static/common/js/user.js:279]: Nice.

File: resources/static/dialog/js/misc/state.js [hunk 2]
```diff
@@ -424,6 +424,12 @@ BrowserID.State = (function() {
         redirectToState("stage_reverify_email", info);
       }
       else {
+        // make sure an unverified-certs are removed
+        if (record.unverified && info.state !== 'unverified') {
+          storage.invalidateEmail(email);
```
Comment (ozten) [inline resources/static/dialog/js/misc/state.js:429]: Very cool.

**Comment by ozten** [comment]
Frontend Tests failing with

```
request for 'get /wsapi/address_info?email=testemail%40testemail.com&issuer=default valid' is not mocked in resources/static/test/cases/xhr.js
```

Restarting Travis Build.

**Comment by seanmonstar** [comment]
@ozten fixed up the tests!

**Pull request merged**